gcs-fetcher/pkg/fetcher/fetcher.go

```diff
@@ -760,20 +762,123 @@ func (gf *Fetcher) fetchFromZip(ctx context.Context) (err error) {
 	return nil
 }
 
+// fetchFromTarGz is used when downloading a single .tar.gz of source files. It
+// is responsible to fetch the .tar.gz file and unzip it into the destination
+// folder.
+func (gf *Fetcher) fetchFromTarGz(ctx context.Context) (err error) {
+	started := time.Now()
+	gf.log("Fetching archive %s.", formatGCSName(gf.Bucket, gf.Object, gf.Generation))
+
+	// Download the archive from GCS.
+	tgzDir := gf.StagingDir
+	j := job{
+		filename:        gf.Object,
+		bucket:          gf.Bucket,
+		object:          gf.Object,
+		generation:      gf.Generation,
+		destDirOverride: tgzDir,
+	}
+	report := gf.fetchObject(ctx, j)
+	if !report.success {
+		return fmt.Errorf("failed to download archive %s: %v", formatGCSName(gf.Bucket, gf.Object, gf.Generation), report.err)
+	}
+
+	// Untgz into the destination directory
+	untgzStart := time.Now()
+	tgzfile := filepath.Join(tgzDir, gf.Object)
+	f, err := os.Open(tgzfile)
+	if err != nil {
+		return err
+	}
+	gzr, err := gzip.NewReader(f)
+	if err != nil {
+		return err
+	}
+	tr := tar.NewReader(gzr)
+
+	defer func() {
+		if cerr := f.Close(); cerr != nil {
+			err = fmt.Errorf("Failed to close file %q: %v", tgzfile, cerr)
```

Comment:
Should this error be logged? Not sure where it gets used later.

Reply:
`fetchFromTarGz` uses a named return value, so this err gets returned by the method.